File: src/main/java/co/uk/jiveelection/campaign/jive/ge2017/MayJive.java
package co.uk.jiveelection.campaign.jive.ge2017;

import co.uk.jiveelection.campaign.TwitConfig;
import co.uk.jiveelection.campaign.input.twitter.TwitterInput;
import co.uk.jiveelection.campaign.jive.Jive;
import co.uk.jiveelection.campaign.output.twitter.TwitterOutput;
import co.uk.jiveelection.campaign.translator.memory.InMemoryJiveTranslator;
import twitter4j.*;
import twitter4j.conf.Configuration;
import twitter4j.conf.ConfigurationBuilder;

public class MayJive extends Jive<TwitterInput, TwitterOutput, InMemoryJiveTranslator> {

    public MayJive() {
        final Configuration configuration = new ConfigurationBuilder()
                .setOAuthConsumerKey(TwitConfig.CONSUMER_TOKEN)
                .setOAuthConsumerSecret(TwitConfig.CONSUMER_TOKEN_SECRET)
                .setOAuthAccessToken(TwitConfig.JIVERESA_MAY_ACCESS_TOKEN)
                .setOAuthAccessTokenSecret(TwitConfig.JIVERESA_MAY_ACCESS_TOKEN_SECRET)
                .build();

        final Twitter jiveMayTwitter = new TwitterFactory(configuration).getInstance();
        final TwitterStream jiveMayTwitterStream = new TwitterStreamFactory(configuration).getInstance();

        input = new TwitterInput(this, "theresa_may", jiveMayTwitter, jiveMayTwitterStream);
        output = new TwitterOutput(this, jiveMayTwitter);
        translator = new InMemoryJiveTranslator(this);
    }

    @Override
    public void init() throws TwitterException {
        input.init();
    }

    @Override
    public void onJiveComplete(Object o) {
        // Do nothing for now.
    }
}

File: src/main/kotlin/co/uk/jiveelection/campaign/jive/ge2017/MayJive.kt
package co.uk.jiveelection.campaign.jive.ge2017

import co.uk.jiveelection.campaign.input.twitter.TwitterInput
import co.uk.jiveelection.campaign.input.twitter.TwitterInputFactory
import co.uk.jiveelection.campaign.jive.Jive
import co.uk.jiveelection.campaign.output.twitter.TwitterOutput
import co.uk.jiveelection.campaign.output.twitter.TwitterOutputFactory
import co.uk.jiveelection.campaign.translator.memory.InMemoryJiveTranslator
import co.uk.jiveelection.campaign.translator.memory.InMemoryJiveTranslatorFactory
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import twitter4j.TwitterException
import javax.inject.Inject

class MayJive @Inject
constructor(
        twitterInputFactory: TwitterInputFactory,
        twitterOutputFactory: TwitterOutputFactory,
        inMemoryJiveTranslatorFactory: InMemoryJiveTranslatorFactory
) : Jive<TwitterInput, TwitterOutput, InMemoryJiveTranslator>() {

    private val logger: Logger = LoggerFactory.getLogger(MayJive::class.java)

    override val input: TwitterInput = twitterInputFactory.create(this)
    override val output: TwitterOutput = twitterOutputFactory.create(this)
    override val translator: InMemoryJiveTranslator = inMemoryJiveTranslatorFactory.create(this)

    @Throws(TwitterException::class)
    override fun init() {
        input.init()
        logger.info("MayJive initialised.")
    }

    override fun onJiveComplete(o: Any) {
        // Do nothing for now.
    }
}
